packages/web3-providers-http/src/index.js

```diff
@@ -19,29 +19,33 @@
  *   Marek Kotewicz <marek@parity.io>
  *   Marian Oancea
  *   Fabian Vogelsteller <fabian@ethereum.org>
+ *   AyanamiTech <ayanami0330@protonmail.com>
  * @date 2015
  */
 
 var errors = require('web3-core-helpers').errors;
-var XHR2 = require('xhr2-cookies').XMLHttpRequest; // jshint ignore: line
+var fetch = require('cross-fetch');
 var http = require('http');
 var https = require('https');
 
+// Apply missing polyfill for IE
+require('es6-promise').polyfill();
+require('abortcontroller-polyfill/dist/polyfill-patch-fetch');
 
 /**
  * HttpProvider should be used to send rpc calls over http
  */
 var HttpProvider = function HttpProvider(host, options) {
     options = options || {};
 
-    this.withCredentials = options.withCredentials || false;
+    this.withCredentials = options.withCredentials;
     this.timeout = options.timeout || 0;
     this.headers = options.headers;
     this.agent = options.agent;
     this.connected = false;
 
     // keepAlive is true unless explicitly set to false
-    const keepAlive = options.keepAlive !== false;
+    var keepAlive = options.keepAlive !== false;
```

Comment:
```suggestion
    const keepAlive = options.keepAlive !== false;
```
User can explicitly set it via options, why we need this change?